src/solvers/dgsem_tree/subcell_limiters_2d.jl

```diff
@@ -291,4 +321,180 @@ end
 
     return nothing
 end
+
+@inline function idp_positivity_nonlinear!(alpha, limiter, u, dt, semi, variable)
+    _, equations, dg, cache = mesh_equations_solver_cache(semi)
+    (; positivity_correction_factor) = limiter
+
+    (; variable_bounds) = limiter.cache.subcell_limiter_coefficients
+    var_min = variable_bounds[Symbol(string(variable), "_min")]
+
+    @threaded for element in eachelement(dg, semi.cache)
+        inverse_jacobian = cache.elements.inverse_jacobian[element]
+        for j in eachnode(dg), i in eachnode(dg)
+            # Compute bound
+            u_local = get_node_vars(u, equations, dg, i, j, element)
+            var = variable(u_local, equations)
+            if var < 0
+                error("Safe $variable is not safe. element=$element, node: $i $j, value=$var")
+            end
```

Comment:
Do you really want to test this in each iteration in production or is this just a debugging leftover?

Reply:
I've got the same question when adding the positivity limiter for conservative variables [here](https://github.com/trixi-framework/Trixi.jl/blob/fcf2652f8ce3f16ba640a66990134632c9f8d3d5/src/solvers/dgsem_tree/subcell_limiters_2d.jl#L247-L249).
I still think it is a nice thing to have for every simulation since when you are receiving this error you can be almost sure that the time step is just too big and reducing the CFL probably fixes the problem.

But of course, checking this for every node in every RK stage is not ideal for the performance. @amrueda what do you think about that? You kept these lines in Fluxo, don't you?

Reply:
I tested to remove this check, and it doesn't seem to have a performance influence. Therefore, I would like to keep such a check if that okay with you @sloede.
But I'm thinking about changing the error message, since it is not really important where the error occurred.
Thinking about something like `error("Safe low-order method produces negative value for variable $variable. Try a smaller time step.")`.
Is something else than a normal `error()` better suited for this check?